### PR TITLE
Fix Gaussian noise: transforms updates & distance noise calculation

### PR DESCRIPTION
**Pull request by msz-rai**

### CODE REVIEW

**Review by PawelLiberadzki — APPROVED** [review]

File: src/gpu/gaussianNoiseKernels.cu [hunk 1]
```diff
@@ -65,7 +65,7 @@ __global__ void kAddGaussianNoiseDistance(size_t pointCount, float mean, float s
 		outDistances[tid] = distance + distanceError;
 	}
 
-	outPoints[tid] = inPoints[tid] + inPoints[tid].normalize() * distanceError;
+	outPoints[tid] = lookAtOriginTransform.inverse() * (originPoint + originPoint.normalize() * distanceError);
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply src/gpu/gaussianNoiseKernels.cu:68]: 1. We are transforming the point to the sensor's origin - line 56
2. We are calculating the distance from XYZ and estimating distance error (Gaussian noise) - lines 58-62
3. We are applying distance error to the point in the sensor's origin and then returning to the original origin - line 68

Before my change, the distance error was applied to the point in the original origin - the direction of the distance error applied was wrong.

File: test/src/gaussianPoseIndependentTest.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#include <helpers/lidarHelpers.hpp>
+#include <helpers/sceneHelpers.hpp>
+
+// The aim of this test is to verify that gaussian noise is not dependent on the ray's pose.
+// The test will compare point clouds captured inside cubes.
+// Cubes are spawned in different locations. Raytrace will always be performed from the same pose in relation to the cube.
+// Gaussian noise is configured with zero standard deviation so there is no randomization.
+
+constexpr float MEAN = 0.1f;
+constexpr float EPSILON_DIFF = 1e-4;
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply test/src/gaussianPoseIndependentTest.cpp]: 1. It is a part of RGL establishment :)
2. Fixed

**Review by prybicki — APPROVED** [review]
LGTM

File: test/src/gaussianPoseIndependentTest.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+#include <helpers/commonHelpers.hpp>
+#include <helpers/lidarHelpers.hpp>
+#include <helpers/sceneHelpers.hpp>
+
+// The aim of this test is to verify that gaussian noise is not dependent on the ray's pose.
+// The test will compare point clouds captured inside cubes.
+// Cubes are spawned in different locations. Raytrace will always be performed from the same pose in relation to the cube.
+// Gaussian noise is configured with zero standard deviation so there is no randomization.
+// The test checks that the noise is always applied correctly in ray coordinates.
+// There was a bug in the past where the point cloud transform was not updating correctly in the Gaussian noise nodes.
```
Comment (msz-rai) [inline test/src/gaussianPoseIndependentTest.cpp:11]:
```suggestion
// There was a bug in the past where the point cloud transform was not updated correctly in the Gaussian noise nodes.
```

**Pull request merged**